lib/grp.gi

```diff
@@ -1129,17 +1136,18 @@ end);
 
 InstallMethod( Exponent,
     "method for abelian groups with generators",
-    [ IsGroup and IsAbelian and HasGeneratorsOfGroup],
+    [ IsGroup and IsAbelian and HasGeneratorsOfGroup ],
     function( G )
+    if not IsFinite(G) then
```

Comment:
Is this the proper way to do this? I recall that @hulpke at similar situations would ask to rather call the function for groups having the ````IsFinite```` property, and maybe do a redispatch afterwards. 

It would make more sense, anyway, to compute the exponent from the abelian invariants, e.g. for infinite abelian groups. Maybe another method could be added (with a redispatch again), for groups having abelian invariants computed?

Reply:
I am not sure that computing the exponent from the abelian invariants really is superior in general. It might be in some case, and not in others -- in any case, if somebody wanted to investigate this as a possible improvement, they are welcome; but for this PR, I wanted to focus on the absolute necessary. I.e. fix or tweak existing code, not rewrite it from scratch or replace it with a completely different method.

As to using redispatch instead: I'll change this PR accordingly.

Reply:
Actually, I am reverting this: the problem is that adding `IsFinite` to the the filter of this method changes its rank, breaking the careful ranking hierarchy. Of course I could try to tweak that, but this is brittle, so I'd rather keep the filters unchanged.

Reply:
Actually, I might not revert it: turns out it is sufficient to add `IfFinite` to one more `Exponent` method...